src/librustc_driver/pretty.rs

```diff
@@ -681,7 +681,12 @@ impl<'a> ReplaceBodyWithLoop<'a> {
                                     _ => None,
                                 });
                                 any_involves_impl_trait(types.into_iter()) ||
-                                any_involves_impl_trait(data.bindings.iter().map(|b| &b.ty))
+                                any_involves_impl_trait(data.constraints.iter().filter_map(|c| {
```

Comment:
What are the AST changes here and below for? I don't see any use sites after the parser and AST validation

Reply:
Just because the AST now has a concept of an associated type constraint (either `=` or `:`) rather than just a binding (`=`), and the data structures have changed.